src/main/java/com/couchbase/intellij/tree/cblite/CBLTreeRightClickListener.java

```diff
@@ -189,62 +267,286 @@ public void actionPerformed(@NotNull AnActionEvent e) {
+                                ActiveCBLDatabase.getInstance().getDatabase()
+                                        .performMaintenance(MaintenanceType.COMPACT);
+                            }
+
+                            if (userObject.getVirtualFile() != null) {
+                                try {
+                                    FileEditorManager fileEditorManager = FileEditorManager.getInstance(project);
+                                    fileEditorManager.closeFile(userObject.getVirtualFile());
+                                } catch (Exception ex) {
+                                    ex.printStackTrace();
+                                    Log.debug("Could not close the file", ex);
+                                }
+                            }
+
+                            DefaultMutableTreeNode parentNode = (DefaultMutableTreeNode) clickedNode.getParent();
+                            SwingUtilities.invokeLater(() -> {
+                                if (parentNode != null) {
+                                    ((DefaultTreeModel) tree.getModel()).removeNodeFromParent(clickedNode);
+                                }
+                                ((DefaultTreeModel) tree.getModel()).nodeStructureChanged(parentNode);
+                            });
+                        } catch (Exception ex) {
+                            Log.error("An error occurred while trying to delete the document " + userObject.getId(),
+                                    ex);
+                            SwingUtilities.invokeLater(() -> Messages.showErrorDialog(
+                                    "Could not delete the document. Please check the logs for more.",
+                                    "Couchbase Plugin Error"));
+                        }
+                    }
+                };
+                deleteDocumentTask.queue();
+            }
+        };
+        actionGroup.add(deleteDocument);
+
+        showPopup(e, tree, actionGroup);
+    }
+
+    private static void handleBlob(Project project, MouseEvent e, DefaultMutableTreeNode clickedNode,
+                                   CBLBlobNodeDescriptor userObject, Tree tree) {
+        DefaultActionGroup actionGroup = new DefaultActionGroup();
+
+        AnAction openBlob = new AnAction("Open Blob") {
             @Override
             public void actionPerformed(@NotNull AnActionEvent e) {
                 try {
-                    CBLTreeHandler.deleteConnection(clickedNode, userObject, tree);
-                } catch (CouchbaseLiteException ex) {
-                    Log.error(ex);
-                    SwingUtilities.invokeLater(() -> Messages.showErrorDialog("Could not delete the connection.", "Couchbase Lite Plugin Error"));
+                    // Assuming blob is a file on disk
+                    File blobFile = new File(userObject.getDigest());
+
+                    if (!Desktop.isDesktopSupported()) {
```

Comment:
this strategy here needs to change. We need to open it with the editor if possible or not show this option at all according to the file type.